aws-ts-eks-hello-world/README.md

```diff
@@ -0,0 +1,148 @@
+[![Deploy](https://get.pulumi.com/new/button.svg)](https://app.pulumi.com/new)
+
+# AWS EKS Cluster
+
+This example deploys an EKS Kubernetes cluster with an EBS-backed StorageClass, and deploys a Kubernetes Namespace and Deployment of NGINX
+into the cluster.
+
+## Deploying the App
+
+To deploy your infrastructure, follow the below steps.
+
+### Prerequisites
+
+1. [Install Pulumi](https://pulumi.io/install)
+2. [Install Node.js 8.11.3](https://nodejs.org/en/download/)
+3. [Configure AWS Credentials](https://pulumi.io/install/aws.html)
+4. [Install `aws-iam-authenticator`](https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html#get-started-kubectl)
+
+### Steps
+
+After cloning this repo, from this working directory, run these commands:
+
+1. Install the required Node.js packages:
+
+    ```bash
+    $ npm install
+    ```
+
+1. Create a new stack, which is an isolated deployment target for this example:
+
+    ```bash
+    $ pulumi stack init
+    ```
+
+1. Set the required configuration variables for this program:
+
+    ```bash
+    $ pulumi config set aws:region us-west-2
+    ```
+
+   We recommend using `us-west-2` to host your EKS cluster as other regions (notably `us-east-1`) may have capacity
+   issues that prevent EKS clusters from creating:
+
+    ```
+    Diagnostics:
+      aws:eks:Cluster: eksCluster
+        error: Plan apply failed: creating urn:pulumi:aws-ts-eks-example::aws-ts-eks::EKSCluster$aws:eks/cluster:Cluster::eksCluster: error creating EKS Cluster (eksCluster-233c968): UnsupportedAvailabilityZoneException: Cannot create cluster 'eksCluster-233c968' because us-east-1a, the targeted availability zone, does not currently have sufficient capacity to support the cluster. Retry and choose from these availability zones: us-east-1b, us-east-1c, us-east-1d
+            status code: 400, request id: 9f031e89-a0b0-11e8-96f8-534c1d26a353
+    ```
+
+    We are tracking enabling the creation of VPCs limited to specific AZs to unblock this in `us-east-1`: pulumi/pulumi-aws-infra#32
+
+1. Stand up the EKS cluster, which will also deploy the Kubernetes Dashboard:
+
+    ```bash
+    $ pulumi up
+    ```
+
+1. After 10-15 minutes, your cluster will be ready, and the kubeconfig JSON you'll use to connect to the cluster will
+   be available as an output. You can save this kubeconfig to a file like so:
+
+    As part of the update, you'll see a few new objects in the output such as: a
+    `Namespace` in Kubernetes to deploy into, and a `Deployment` resource for
+    the NGINX app. Pulumi understands which changes to a given cloud
+    resource can be made in-place, and which require replacement, and computes
+    the minimally disruptive change to achieve the desired state.
+    
+    **Note:** Pulumi auto-generates a suffix for all objects. Pulumi's object model does
```

Comment:
Do you think we have to reference this here?    In particular, how does this relate to the rest of the walkthrough?

Reply:
Yeah, I think it would be better to move the lengthier description to https://pulumi.io/reference/programming-model.html and just keep the first sentence with a link to the lengthier description.
```suggestion
    **Note:** By default, Pulumi [auto-generates a suffix for all objects](https://pulumi.io/reference/programming-model.html).
```